include/fi_list.h

```diff
@@ -179,7 +179,7 @@ static inline struct slist_entry *slist_remove_head(struct slist *list)
 }
 
 #define slist_foreach(list, item, prev) \
-	for (prev = NULL, item = list->head; item; prev = item, item = item->next)
+	for ((prev) = NULL, (item) = (list)->head; (item); (prev) = (item), (item) = (item)->next)
```

Comment:
Can you break these macros up into multiple lines that aren't much more then 80 columns?  Yes, I still occasionally read code using vi in a small terminal window.